### PR TITLE
add notes for n25515 

### PR DESCRIPTION
**Pull request by LarryRuane**
@glozow @dergoegge 
I'm pretty happy with the notes but suggestions of course are welcome. The questions could possibly be added to or otherwise improved. Thanks to Niklas for some of the questions, and thanks to both of you for any comments, I welcome nits, nothing is too small! If you have permission to push, feel free to do so if I don't seem to be around (you can try and ping me on slack), I know I'm late with this, sorry.

### CODE REVIEW

**Review by glozow — COMMENTED** [review]
Pinging @dergoegge to see if we can dig a little deeper into the code in the questions, but LGTM.

I've updated the links to point to the bitcoin-core-review-club/bitcoin repo, as the links to the PR commits will disappear if/when Niklas force-pushes again.

**Comment by dergoegge** [comment]
Compressed the notes into fewer bullet points and added a note for the functional tests.

Would be nice to have a question or two about the added unit tests. @LarryRuane Do you think you can come up with some?

**Comment by LarryRuane** [comment]
@dergoegge 
> Do you think you can come up with some?

Excellent idea, I'll work on that right now, thanks.

**Review by stickies-v — APPROVED** [review]
ACK aaa6054 - thanks Larry!

Pushed one more commit to use `bitcoin-core-review-club` links and a slight rephrasing.

File: _posts/2022-10-19-#25515.md [hunk 1]
```diff
@@ -7,18 +7,69 @@ authors: [dergoegge]
...
+  Additionally, when a `PeerManager` instance is created, we now pass a
+  `ConnectionsInterface` instead of a `CConnman`. With that, any unit tests can
+  create a mocked version of the connection manager to test the internals of
+  `PeerManager`.
+
+- To demonstrate the added benefit, the PR also adds two initial `PeerManager`
+  unit tests for the `version` and `ping/pong` messages (see
+  [`peerman_tests.cpp`](https://github.com/bitcoin-core-review-club/bitcoin/blob/f98a4e8d891dd7374ef7dc4c723797bf0705075f/src/test/peerman_tests.cpp)).
+
+- `PeerManager` is currently almost exclusivly tested through our functional
```
Comment (glozow) [inline _posts/2022-10-19-#25515.md:34]:
```suggestion
- `PeerManager` is currently almost exclusively tested through our functional
```

**Pull request merged**